game-core/src/main/java/games/strategy/engine/data/changefactory/units/UnitHitsChange.java

```diff
@@ -0,0 +1,57 @@
+package games.strategy.engine.data.changefactory.units;
+
+import games.strategy.engine.data.Change;
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.Unit;
+import java.util.Collection;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import org.triplea.java.collections.IntegerMap;
+
+/** A game data change that captures the damage done to a collection of units. */
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
+public class UnitHitsChange extends Change {
+  private static final long serialVersionUID = 2862726651812142713L;
+
+  private final IntegerMap<String> newHits;
```

Comment:
It's missing that the `String` represents a unit id. A quick comment like `// unitId -> total damage taken` could do the trick to document that. WDYT?